src/components/Property/RepairsFinder/RepairsFinderInput.tsx

```diff
@@ -1,194 +1,189 @@
-
-      // timeout makes it look cooler
-      setTimeout(() => {
-        setIsLoading(() => false)
-        setRepairsApiResponse(() => response)
-      }, 2000)
-    } catch (e) {
-      console.error(e.message)
-      setIsLoading(() => false)
-      setError('Something went wrong validating sorCode. Please try again')
-    }
-  }
-
-  const onSetXmlCode = (e) => {
-    e.preventDefault()
-
-    setXmlContent(() => e.target.value)
-  }
+    if (matchingSorCode == null) return
+
+    const totalCost =
+      matchingSorCode?.sorCode?.cost * parseInt(extractedXmlData?.quantity)
+
+    setTotalCost(totalCost)
+    setContractorReference(extractedXmlData.contractorReference)
+    setTradeCode(matchingSorCode.tradeCode)
+  }, [matchingSorCode])
 
   return (
     <>
+      <WarningInfoBox
+        className="variant-warning govuk-!-margin-bottom-4"
+        header="Looking to use Repairs Finder?"
+        name="despatched-warning"
+        text={
+          <>
+            Visit{' '}
+            <a
+              className="lbh-link"
+              target="_blank"
+              href="https://product-test.necsws.com/cgi-bin/hackney_rftr_launchalone.pl"
+              rel="noreferrer"
+            >
+              Repairs Finder
+            </a>{' '}
+            to diagnose the repair, then paste the code below.
+          </>
+        }
+      />
+
       <TextArea
         name="xmlContent"
-        value={xmlContent}
         label="Repairs finder code"
         hint="Please paste the code from Repairs Finder"
-        required
-        error={!extractedXmlData && { message: 'Invalid code format' }}
-        onInput={onSetXmlCode}
+        required={true}
+        register={register({
+          required: 'Please enter a code',
+          validate: () => {
+            return error || true
+          },
+        })}
+        error={errors && errors?.xmlContent}
+        onInput={(e) => setTextInput(e.target.value)}
         rows={6}
       />
 
-      {error && <ErrorMessage label={error} />}
-
       <div>{isLoading && <SpinnerWithLabel label="Validating code.." />}</div>
 
+      {matchingSorCode?.hasPropertyContract === false && (
+        <WarningInfoBox
+          className="variant-error govuk-!-margin-bottom-4"
+          header="Repair cannot be raised on this property"
+          name="despatched-warning"
+          text={`The selected work order cannot be raised against this property. `}
+        />
+      )}
+
       <Table className="original-tasks-table">
         <TBody>
           <tr className="govuk-table__row">
             <td className="govuk-table__cell">Trade</td>
             <td className="govuk-table__cell">
-              {repairsApiResponse === null
+              {matchingSorCode === null
                 ? '-'
-                : `${repairsApiResponse?.trade} - ${repairsApiResponse?.tradeCode}`}
+                : `${matchingSorCode?.trade} - ${matchingSorCode?.tradeCode}`}
```

Comment:
Am I correct to understand that if `matchingSorCode` is not missing, then it will 100% contain the `trade` and `tradecode`? Meaning, the `?` is just a fomality? Or perhaps, fallback values should be set to these fields when they're missing?

Same comment for all of the similar logic bellow.